test/Microsoft.AspNetCore.Mvc.Localization.Test/HtmlLocalizerTest.cs

```diff
@@ -158,6 +159,32 @@ public void HtmlLocalizer_HtmlWithInvalidResourceString_ContentThrowsException(s
             Assert.Equal("Input string was not in a correct format.", exception.Message);
         }
 
+        /// <summary>
+        /// This tests that LocalizedString instances containing curly braces (e.g. JSON or code snippets) can be used without a FormatException when no arguments are provided.
+        /// </summary>
+        [Fact]
+        public void HtmlLocalizer_HtmlWithJsonText_ReturnsLocalizedHtmlString()
+        {
+            // Arrange
+            var localizedString = new LocalizedString("ExampleJson", "example JSON: {\"foo\":\"bar\"}");
```

Comment:
Thanks for the hint,  I refactored the whole test class to use it (instead of mocking) and noticed that the existing HtmlLocalizer_UseIndexerWithArguments_ReturnsLocalizedHtmlString test wasn't asserting the arguments.  this is now also fixed.

I kept HtmlLocalizer_HtmlWithArguments_ReturnsLocalizedHtml with Mocking, because I think it's easier to read with the Theory data in HtmlData.